src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7840,21 +7846,25 @@ private BoundExpression BindIndexedPropertyAccess(BoundPropertyGroup propertyGro
             }
 
             var arguments = AnalyzedArguments.GetInstance();
+            Debug.Assert(receiverOpt is not null);
```

Comment:
I'm not sure why this assert should hold true for this branch and not the other one. Can this assert be moved to the start of the method?